src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -10052,16 +10052,36 @@ namespace SurfaceGeometry {
 				Triangle2( 3 ) = vertex( 4 );
 
 				// get total Area of quad.
-				Real64 const TotalArea( surface.GrossArea );
+				Real64 TotalArea( surface.GrossArea );
 				if ( TotalArea <= 0.0 ) {
 					//catch a problem....
 					ShowWarningError( "CalcSurfaceCentroid: zero area surface, for surface=" + surface.Name );
 					continue;
 				}
 
 				// get area fraction of triangles.
-				Real64 const Tri1Area( AreaPolygon( 3, Triangle1 ) / TotalArea );
-				Real64 const Tri2Area( AreaPolygon( 3, Triangle2 ) / TotalArea );
+				Real64 Tri1Area( AreaPolygon( 3, Triangle1 ) / TotalArea );
+				Real64 Tri2Area( AreaPolygon( 3, Triangle2 ) / TotalArea );
+
+				// check if sum of fractions are slightly greater than 1.0 which is a symptom of the triangles for a non-convex quadralateral using the wrong two triangles
+				if ( (Tri1Area + Tri2Area) > 1.05 ) {
```

Comment:
@JasonGlazer What's the basis for >1.05 as the test?  Could a bad surface slip through with, say 1.04?